src/utils/protomessage.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include <cstdint>
+#include <cstring>
+#include <functional>
+#include <map>
+#include <string_view>
+#include <vector>
+
```

Comment:
Just a reminder, it needs to `#include <string>` so it can pass cross-compilation.
Also lots of warnings are silenced (not just for cross-compilation) with 
```
#undef minor
#undef major
```

Reply:
Done!